### PR TITLE
[Models] Fixed wrong imported name

### LINKED ISSUE #992: Bug with previous "Simplify full namespaces for already included resources" commit
This recent [commit](https://github.com/barryvdh/laravel-ide-helper/commit/6f20800eef08db1d6da7aafd383309fcfafc651a) breaks lots of code. Instead of the FQCN, the IDE helper now generates a class alias. Eg, instead of this:

```php
 * @method static \App\Context\Product\QueryBuilders\EMaterialQueryBuilder|\App\Context\Product\Models\EMaterial newModelQuery()
 * @method static \App\Context\Product\QueryBuilders\EMaterialQueryBuilder|\App\Context\Product\Models\EMaterial newQuery()
 * @method static \App\Context\Product\QueryBuilders\EMaterialQueryBuilder|\App\Context\Product\Models\EMaterial query()
```

The following is generated

```php
* @method static EMaterialQueryBuilder|EMaterial newModelQuery()
 * @method static EMaterialQueryBuilder|EMaterial newQuery()
 * @method static EMaterialQueryBuilder|EMaterial query()
```

The same problem happens for example with custom casts:

```php
 * @property \App\Context\Product\Enums\EMaterialType $type
```

Becomes

```php
 * @property EMaterialType $type
```

### PR DESCRIPTION
**Pull request by LANGERGabriel**
Fixes #992 

Disabled generating models phpdoc using imported name in external file, except for the models own class name.

### CODE REVIEW

File: tests/Console/ModelsCommand/GeneratePhpdocWithFqnInExternalFile/Builders/EMaterialQueryBuilder.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
```
Comment (mfn) [inline tests/Console/ModelsCommand/GeneratePhpdocWithFqnInExternalFile/Builders/EMaterialQueryBuilder.php:1]: May I suggest we (this package) always tries to use `declare(strict_types=1);`, at least in all the tests? thanks!

File: src/Console/ModelsCommand.php [hunk 2]
```diff
@@ -1051,24 +1057,31 @@ protected function getTypeInModel(object $model, ?string $type): ?string
...
+    protected function getClassNameInDestinationFile(object $model, string $className): string
     {
+        $reflection = $model instanceof ReflectionClass
+            ? $model
+            : new ReflectionObject($model)
+        ;
+
         $className = trim($className, '\\');
-        $usedClassNames = $this->getUsedClassNames($model);
+        $writingToExternalFile = !$this->write;
```
Comment (mfn) [inline src/Console/ModelsCommand.php:1068]: ❤️ for temp var with meaningful name

File: tests/Console/ModelsCommand/GeneratePhpdocWithFqnInExternalFile/Builders/EMaterialQueryBuilder.php (hunk 1 above)
Comment (mr-feek) [reply tests/Console/ModelsCommand/GeneratePhpdocWithFqnInExternalFile/Builders/EMaterialQueryBuilder.php:1]: Let's add a phpcs rule for this then! https://github.com/slevomat/coding-standard#slevomatcodingstandardtypehintsdeclarestricttypes-

We can configure it just for the tests dir

**Comment by barryvdh** [comment]
@brendt does this solve your issue?

File: tests/Console/ModelsCommand/GeneratePhpdocWithFqnInExternalFile/Builders/EMaterialQueryBuilder.php (hunk 1 above)
Comment (LANGERGabriel) [reply tests/Console/ModelsCommand/GeneratePhpdocWithFqnInExternalFile/Builders/EMaterialQueryBuilder.php]: Declare added

Comment (mfn) [reply tests/Console/ModelsCommand/GeneratePhpdocWithFqnInExternalFile/Builders/EMaterialQueryBuilder.php]: @mr-feek 🤗

**Comment by barryvdh** [comment]
So this is good to merge?

**Comment by brendt** [comment]
Looking good! 👍

**Comment by barryvdh** [comment]
Can you test with `2.7.x@dev`?

**Comment by brendt** [comment]
Yes, works!

**Comment by brendt** [comment]
Thanks Barry and Gabriel

**Pull request merged**